specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/applications.json

```diff
@@ -324,7 +331,15 @@
         },
         "applicationType": {
           "type": "string",
-          "description": "The application type."
+          "description": "The application type.",
+          "enum": [
+            "CustomApplication",
+            "RServer"
+          ],
+          "x-ms-enum": {
+            "name": "ApplicationType",
+            "modelAsString": true
+          }
```

Comment:
breaking change.

Reply:
> breaking change.

Hi anuchandy, Why do you think it is a breaking change? 
I have tested  the generated .NET SDK based this swagger spec. In .NET SDK, it is not a breaking change. Because the `modelAsString` is true, it will change it to enum in .NET SDK
Can you give me a example that confirm that it will bring up a breaking change.

Reply:
IMO this is only a breaking change if server can return a value (or allow setting a value) that isn't specified in this list. Otherwise, I don't see a problem. @aim-for-better please confirm.

Reply:
> IMO this is only a breaking change if server can return a value (or allow setting a value) that isn't specified in this list. Otherwise, I don't see a problem. @aim-for-better please confirm.

@antmarti-microsoft  Thanks for you opinion. Now Rp server side only support these two types. So I think there is no problem.

Reply:
@aim-for-better - the scope of breaking change is different for different languages, for example in java this cause generating a model that extends from ExpandableStringEnum with one value WebPage, like [this](https://github.com/Azure/azure-sdk-for-java/blob/18e9a4b927fafc64349de742ccd2f546da9c653b/compute/resource-manager/v2017_03_30/src/main/java/com/microsoft/azure/management/compute/v2017_03_30/IPVersion.java#L18). I agree that this is not a breaking change in API Level.

Reply:
> @aim-for-better - the scope of breaking change is different for different languages, for example in java this cause generating a model that extends from ExpandableStringEnum with one value WebPage, like [this](https://github.com/Azure/azure-sdk-for-java/blob/18e9a4b927fafc64349de742ccd2f546da9c653b/compute/resource-manager/v2017_03_30/src/main/java/com/microsoft/azure/management/compute/v2017_03_30/IPVersion.java#L18). I agree that this is not a breaking change in API Level.

Hi @anuchandy  I have tested the java code, I think you are right. However the PR has been merged. So I will pull a new request to fix this breaking change.